README.md

```diff
@@ -52,6 +52,7 @@ Some examples require extra dependencies. See each sample's directory for specif
   * [hello_signal](hello/hello_signal.py) - Send signals to a workflow.
 <!-- Keep this list in alphabetical order -->
 * [activity_worker](activity_worker) - Use Python activities from a workflow in another language.
+* [atomic_message_handlers](updates_and_signals/atomic_message_handlers/) - Safely handling updates and signals.
```

Comment:
I think the name of the sample should be changed to something like `safe_message_handling`. It's not about atomicity -- the sample doesn't demonstrate rolling back of incomplete side effects. Rather it's about maintaining strict isolation between handler executions, via serialization of handler executions. In any case, we don't want users to think this is showing a specialized form of message handling that they can ingnore; we want them to consider whether they need this for any workflow with message handlers.

Reply:
Good idea.  "Safe" feels much more like something I'm supposed to read.

Reply:
Not the biggest fan of "safe" vs "atomic" since the latter is more discoverable/descriptive when looking at the list of samples, but I don't have a strong opinion here.

Reply:
@cretz we could choose a word other than "safe", but I argued above that "atomic" isn't the right word.

Reply:
I don't think "atomic" relates to rollback at all. Atomic just means one at a time, as opposed to "transactional". I think atomic is a good word, but again I don't have a strong opinion.

Reply:
Yes, to be honest I'm not in love with "safe" and implying that any other usage style is not safe.

Hm, an atomic operation is one that either completes in its entirety or behaves as if it never started, and can't be seen in an intermediate state. So, if the operation has multiple stages with side effects, that would require some notion of rollback. It's usually synonymous with "transactional". I agree it's closely related to the idea of serializing executions so that they occur one at a time, since that's one way of ensuring that one execution can't see in-progress state of another, but using "atomic" would imply that message handling that does multiple writes can rollback incomplete changes. I think here we're talking about "serialized message processing" or "preventing corruption of shared state by message handlers".

Reply:
There's more here than just concurrency, such as dangling handlers.  Sticking with safe.
I'm think I'm going to touch on idempotency as well in my next push, though we probably should also add a more focused idempotency sample.

Reply:
> Yes, to be honest I'm not in love with "safe" and implying that any other usage style is not safe.

I don't think it implies that.  "Robust" is an alternate word.

Reply:
Update: added idempotency.  I didn't use the built-in update ID, though I could